lib/yaml_normalizer.rb

```diff
@@ -37,20 +33,34 @@ def handle_array(array)
 
   def handle_value(value)
     if value.is_a?(String)
-      trim(value)
+      handle_string(value)
     elsif value.is_a?(Array)
       handle_array(value)
     elsif value.is_a?(Hash)
       handle_hash(value)
-    elsif value && value != true && value != false
+    elsif value == true || value == false
+      value
+    elsif value
       raise ArgumentError, "unknown YAML value #{value}"
     end
   end
 
+  def handle_string(value)
+    [
+      :trim,
+      :format_punctuation,
+    ].reduce(value) { |result, method| send(method, result) }
+  end
```

Comment:
If we use `Method` objects, we get an immediate error trying to get the method (if they get renamed/deleted), rather than inside the loop?

```suggestion
  def handle_string(value)
    [
      method(:trim),
      method(:format_punctuation),
    ].reduce(value) { |result, method| method.call(result) }
  end
```

Reply:
Hm, yeah, I could see that being a good thing to surface. In an earlier iteration I'd taken a different approach which is somewhat similar in composing the result as a chain of methods:

https://github.com/18F/identity-idp/blob/368df979881fb2663ddb0a24a079bbeea18a181d/lib/yaml_normalizer.rb#L46-L49

I liked the idea of using [`Method#>>`](https://ruby-doc.org/core-3.0.1/Method.html#method-i-3E-3E) for composition, though it felt a bit "clever" and not quite so ergonomic to scale to include more methods.

I think the `reduce` suggestion you propose reads a bit easier or is at least more familiar. I'll go with that 👍

Reply:
> I think the `reduce` suggestion you propose reads a bit easier or is at least more familiar. I'll go with that 👍

Implemented in the rebased 4f07697.